### PR TITLE
feat(fetch): fulfill without passing fetch response body client<->server

### PR DESCRIPTION
**Pull request by yury-s**
#5999 
#1774

### CODE REVIEW

File: src/client/page.ts [hunk 1]
```diff
@@ -437,6 +438,10 @@ export class Page extends ChannelOwner<channels.PageChannel, channels.PageInitia
     return this._mainFrame.evaluate(pageFunction, arg);
   }
 
+  async _fetch(url: string, options: { url?: string, method?: string, headers?: Headers, postData?: string | Buffer, timeout?: number } = {}): Promise<network.FetchResponse> {
```
Comment (dgozman) [inline src/client/page.ts:441]: Perhaps we should make a type for FetchOptions?

File: src/server/network.ts [hunk 2]
```diff
@@ -219,11 +219,18 @@ export class Route extends SdkObject {
     await this._delegate.abort(errorCode);
   }
 
-  async fulfill(overrides: { status?: number, headers?: types.HeadersArray, body?: string, isBase64?: boolean, useInterceptedResponseBody?: boolean }) {
+  async fulfill(overrides: { status?: number, headers?: types.HeadersArray, body?: string, isBase64?: boolean, useInterceptedResponseBody?: boolean, fetchResponseUid?: string }) {
     assert(!this._handled, 'Route is already handled!');
     this._handled = true;
     let body = overrides.body;
     let isBase64 = overrides.isBase64 || false;
+    if (!body && overrides.fetchResponseUid) {
```
Comment (dgozman) [inline src/server/network.ts:227]: `!body` => `body === undefined`? You do the same below, I guess it breaks with empty body?

**Review by dgozman — APPROVED** [review]

File: src/server/network.ts (hunk 2 above)
Comment (yury-s) [reply src/server/network.ts:227]: Done.

File: src/client/page.ts (hunk 1 above)
Comment (yury-s) [reply src/client/page.ts:441]: Done.

**Pull request merged**